packages/bsky/src/data-plane/client/index.ts

```diff
@@ -0,0 +1,123 @@
+import assert from 'node:assert'
```

Comment:
Wish this file turned-up better in the diff!  It is largely the contents of `packages/bsky/src/data-plane/client.ts` though there are some alterations so that it works with a `HostList` rather than a static list of host strings.

One change here to the retry logic is that we previously would attempt to retry on a separate host ip if any host on that ip failed.  I think this was a legacy behavior that we haven't needed for some time, and especially with etcd we'd like to defer to its host list.

Reply:
We would attempt the retry on a different host ip.

So e.g. if the four hosts are:
```
http://192.168.1.1:3000
http://192.168.1.1:3001
http://192.168.1.2:3000
http://192.168.1.2:3001
```

And a request to `http://192.168.1.1:3001` failed.  Then we would retry hosts with a different ip, i.e. one of:
```
http://192.168.1.2:3000
http://192.168.1.2:3001
```